### PR TITLE
Remove unittest dependency and test fixtures

### LINKED ISSUE #318: Overhaul unittest and fixtures
Remove dependency on `unittest.TestCase` by replacing assertions with `assert` statements. This also involves using `pytest.raises` and `pytest.skip` decorators. 

Use PyTest fixtures for datasets and other fixtures. 

~Add PEP checking to pytest (alongside pyflakes)~

- [x] rename `ThingTests` to `TestThing` 
- [x] assertions 
- [x] pytest.skip 
- [x] dataset fixtures 
- [ ] ~PEP8~

Alongside this we should also do a PY2 removal check, namely anywhere:

```python
try:
    from unittest import mock
except ImportError:
    import mock
```

should be replaced with just `from unittest import mock`

### PR DESCRIPTION
**Pull request by bbengfort**
This PR fixes #318 - a contagious issue that begins to remove our `unittest` dependencies.

1. Modified `VisualTestCase` to no longer extend `unittest.TestCase`
2. Changed `VisualTestCase.setUp` to `VisualTestCase.setup_method`
3. Added a `pytest_configure` to `conftest.py` for pre-test configurations (needs work)
4. Updated the documentation to remove references to `unittest`
5. Updated assertions to use `assert` rather than `self.assert`
6. Updated `unittest.skip` to `pytest.mark.skip`
7. Updated test cases that subclass `unittest.TestCase` to extend `object` instead

### TODOs and questions

Still to do:

- [x] Check that we didn't lose any tests (e.g. via Test suffix instead of prefix)
- [x] Update TODO: parameterize tests 
- [ ] ~Rename `tests.fixtures.TestDataset` to `tests.fixtures.Dataset` (prefix warnings in runner) (possibly add to #682 )~
- [ ] ~Remove `DatasetMixin` and dataset.py (possibly do in #682)~
- [x] Ensure that CI is still working with new tests
- [x] Ensure finalize is called instead of poof 
- [ ] ~Review warnings issued by tests and docs builds (#707)~
- [ ] ~Update tests to use `windows_tol` (add to #682)~

Questions for the @DistrictDataLabs/team-oz-maintainers:

- Do we want `self.assert_images_similar` or `assert_images_similar`? 
- How do we better use pytest fixtures to ensure the test environment is correct? E.g. instead of xunit style setup and teardown. 

### CHECKLIST

- [x] _Is the commit message formatted correctly?_
- [ ] _Have you noted the new functionality/bugfix in the release notes of the next release?_

- [ ] ~_Included a sample plot to visually illustrate your changes?_~
- [ ] ~_Do all of your functions and methods have docstrings?_~
- [x] _Have you added/updated unit tests where appropriate?_
- [x] _Have you updated the baseline images if necessary?_
- [x] _Have you run the unit tests using `pytest`?_
- [ ] ~_Is your code style correct (are you using PEP8, pyflakes)?_~
- [ ] ~_Have you documented your new feature/functionality in the docs?_~

- [x] _Have you built the docs using `make html`?_

### CODE REVIEW

**Comment by bbengfort** [comment]
In order to ensure that the collected tests are similar or are changing in expected ways, we printed out all of the collected tests from PyTest as follows:

```
$ pytest --collect-only >> collected_tests.txt
```

This file can then have all the warnings and coverage material deleted from it, and it is then easy to diff two collected tests reports to determine what has changed. Based on a diff of the files generated from the original and from the overhaul tests, I'm confident that we did not lose any tests (but added a few).

File: tests/test_features/test_rfecv.py [hunk 1]
```diff
@@ -164,17 +164,17 @@ def test_numpy_integration(self):
         cv = StratifiedKFold(n_splits=4, random_state=32)
         oz = RFECV(RandomForestClassifier(random_state=83), cv=cv)
         oz.fit(X, y)
-        oz.poof()
+        oz.finalize()
 
         self.assert_images_similar(oz, remove_legend=True)
 
-    def test_invalid_step(self):
+    @pytest.mark.parametrize("step", [0, -1, -5])
```
Comment (rebeccabilbro) [inline tests/test_features/test_rfecv.py]: Ooh, another nice use of parametrize! cc: @ndanielsen @pdamodaran @Kautumn06

File: tests/test_regressor/test_alphas.py [hunk 2]
```diff
@@ -57,33 +59,35 @@ def test_similar_image(self):
 
         X, y = make_regression(random_state=0)
         visualizer.fit(X, y)
-        visualizer.poof()
+        visualizer.finalize()
 
         self.assert_images_similar(visualizer)
 
-    def test_regressor_cv(self):
+    @pytest.mark.parametrize("model", [SVR, Ridge, Lasso, LassoLars, ElasticNet])
```
Comment (rebeccabilbro) [inline tests/test_regressor/test_alphas.py]: Thanks for parametrizing the models!

File: tests/test_regressor/test_alphas.py [hunk 3]
```diff
@@ -57,33 +59,35 @@ def test_similar_image(self):
...
+        with pytest.raises(YellowbrickTypeError):
+            AlphaSelection(model())
 
-        # TODO: parametrize with models when unittest dependency removed (new test case)
-        for model in (RidgeCV, LassoCV, LassoLarsCV, ElasticNetCV):
-            try:
-                AlphaSelection(model())
-            except YellowbrickTypeError:
-                pytest.fail("could not instantiate RegressorCV on alpha selection")
+    @pytest.mark.parametrize("model", [RidgeCV, LassoCV, LassoLarsCV, ElasticNetCV])
```
Comment (rebeccabilbro) [inline tests/test_regressor/test_alphas.py]: :+1:

File: tests/test_style/test_palettes.py [hunk 4]
```diff
@@ -188,18 +188,19 @@ def test_other_palettes(self):
         pals = ["flatui", "paired", "neural_paint", "set1"]
         for name in pals:
             pal_out = color_palette(name)
-            self.assertTrue(pal_out)
+            assert pal_out is not None
+            assert len(pal_out) > 0
```
Comment (rebeccabilbro) [inline tests/test_style/test_palettes.py]: 👍

File: tests/test_target/test_binning.py [hunk 5]
```diff
@@ -11,29 +11,28 @@
...
+        # Load the data from the fixture
+        dataset = self.load_data('occupancy')
+
+        # Get the data
+        y = dataset["temperature"]
+
+        visualizer = BalancedBinningReference()
+        visualizer.fit(y)
+        visualizer.finalize()
+        self.assert_images_similar(visualizer, tol=0.5)
```
Comment (rebeccabilbro) [inline tests/test_target/test_binning.py]: Good catch of this extra indentation!

File: yellowbrick/features/rfecv.py [hunk 6]
```diff
@@ -171,7 +171,7 @@ def fit(self, X, y=None):
         else:
             step = int(self.step)
 
-        if step < 0:
+        if step <= 0:
```
Comment (rebeccabilbro) [inline yellowbrick/features/rfecv.py]: 👍

**Pull request merged**